osu.Game/Rulesets/Filter/FilterCriteria_QueryParser.cs

```diff
@@ -6,75 +6,121 @@
 using System.Text.RegularExpressions;
 using osu.Game.Beatmaps;
 
-namespace osu.Game.Screens.Select
+namespace osu.Game.Rulesets.Filter
 {
-    internal static class FilterQueryParser
+    public partial class FilterCriteria
     {
         private static readonly Regex query_syntax_regex = new Regex(
-            @"\b(?<key>stars|ar|dr|hp|cs|divisor|length|objects|bpm|status|creator|artist)(?<op>[=:><]+)(?<value>("".*"")|(\S*))",
+            @"\b(?<key>\w+)(?<op>(:|=|(>|<)(:|=)?))(?<value>("".*"")|(\S*))",
```

Comment:
Explanation for the regex changes:

* `key` is now any sequence of word characters (whether or not they'll get stripped from the query is up to `tryParseKeywordCriteria`)
* For the `op` part - it is:

```
:           single colon
|           or
=           single equals sign
|           or
(>|<)       a greater-than or less-than sign, followed by
(:|=)?      possibly a single colon or a single equals sign
```

Therefore the intention is to allow only `:`. `=`. `>`, `<`, `>=`, `<=`, `>:`, and `<:`, exactly. The previous logic would allow an any-length sequence of characters from the set of `=:><`.